packages/material-ui/src/Portal/Portal.d.ts

```diff
@@ -2,10 +2,26 @@ import * as React from 'react';
 import { PortalProps } from '../Portal';
 
 export interface PortalProps {
+  /**
+   * The children to render into the `container`
+   */
   children: React.ReactElement;
+  /**
+   * A node, component instance, or function that returns either.
+   * The `container` will have the portal children appended to it.
+   * By default, it uses the body of the top-level document object,
+   * so it's simply `document.body` most of the time.
+   */
   container?: React.ReactInstance | (() => React.ReactInstance) | null;
```

Comment:
@eps1lon Do you know what `React.ReactInstance` (resolves to `React.Component | Element`) is in PropTypes?
Perhaps there is a better type for this

Reply:
Does `PropTypes.oneOfType([PropTypes.instanceOf(React.Component), PropTypes.instanceOf(Element)])` work? `Element` might be a bit tricky for node environments though.

Reply:
Would require too much work upstream, and doesn't seem right.
Looking at https://www.npmjs.com/package/prop-types I think it's `PropTypes.element`

Reply:
> I think it's PropTypes.element

That would be the returnval of `React.createElement`

>  Would require too much work upstream, and doesn't seem right.

What makes you say that? If you expect `container: Element` then `element instanceof Element` is true therefore `PropTypes.instanceOf(Element)` is fitting. If you expect `container: React.Component` (i.e. a valid argument for `findDOMNode`) then `container instanceof React.Component` is true therefore `PropTypes.instanceOf(React.Component)` is fitting. If you expect a union of those types `PropTypes.oneOfType` is fitting.

> Would require too much work upstream

How so? The simplest transformer would take every type and convert it to `PropTypes.instanceOf(type)`. `PropTypes.number` is almost equivalent to `PropTypes.instanceOf(Number)`.

Reply:
> What makes you say that?

Doesn't seem like an uncommon pattern, was thinking it was already built into `PropTypes`

> How so?

I was thinking more in the way of: I have no way (as far as I can tell) to check if a type/interface with the name `Element` is the global one or one set by a user.

I ended up ignoring that concern and just added it, so it uses `PropTypes.instanceOf(Element)` and `PropTypes.instanceOf(Element)` now.

Reply:
Fundamentally `PropTypes.instanceOf` isn't something that exists in typescript since it checks structurally. But translating every interface into `PropTypes.shape` isn't helpful either.

> Yeah, tests are currently failing on it

Does `typeof Element === 'undefined' ? PropTypes.fail : PropTypes.instanceOf(Element)` work?

Reply:
`yarn docs:api` works, though I do see a warning
```
Warning: Invalid argument supplied to oneOfType. Expected an array of check functions, but received undefined at index 1.
```

The docs generator probably needs to be updated
![image](https://user-images.githubusercontent.com/3842800/61591876-0f4ea080-abcc-11e9-8607-c6b579eadada.png)

Reply:
Could probably do this to get rid of the warning
```
PropTypes.instanceOf(typeof Element === 'undefined' ? Object : Element)
```

Reply:
> Yeah, tests are currently failing on it

I meant ci is failing on it, specifically `yarn docs:api` and `yarn docs:dev`

Reply:
Updated `generateMarkdown` to handle `instanceOf`.
And used `PropTypes.instanceOf(typeof Element === 'undefined' ? Object : Element)` for `Element`

Reply:
I just wanted to chime in and say that `instanceof` can be problematic in multi-window scenarios (e.g. `window.open()` will have its own `Element`, `Array` etc): https://github.com/reduxjs/react-redux/pull/628.